src/core/annotation.js

```diff
@@ -1251,6 +1255,15 @@ class WidgetAnnotation extends Annotation {
     });
     data.defaultFieldValue = this._decodeFormValue(defaultFieldValue);
 
+    if (fieldValue === undefined && params.xfaDatasets) {
+      // Try to figure out if we have something in the xfa dataset.
+      const path = stringToPDFString(dict.get("T") || "");
+      if (path) {
+        data.hasValueFromXFA = true;
```

Comment:
Please note that by adding a value to the `data` object you're exposing it in the API as well.

Given that we're only using this in the worker-thread, can we please store (and access) this on the class-instance as `this._hasValueFromXFA = true;` instead?